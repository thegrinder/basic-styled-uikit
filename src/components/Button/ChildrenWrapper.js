import { bool } from 'prop-types';
import styled from 'styled-components';

const propTypes = {
  submitting: bool.isRequired,
};

const ChildrenWrapper = styled.span`
  display: flex;
  align-items: center;
  justify-content: space-between;
  opacity: ${({ submitting }) => (submitting ? '0' : '1')};
`;

ChildrenWrapper.propTypes = propTypes;

export default ChildrenWrapper;
