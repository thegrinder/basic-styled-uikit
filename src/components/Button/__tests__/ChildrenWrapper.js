import React from 'react';
import { render, cleanup } from 'react-testing-library';
import 'jest-dom/extend-expect';

import ChildrenWrapper from '../ChildrenWrapper';

describe('<ChildrenWrapper />', () => {
  afterEach(cleanup);

  it('should render correctly', () => {
    const { container: { firstChild } } = render(
      <ChildrenWrapper submitting={true} />,
    );
    expect(firstChild).toBeDefined();
    expect(firstChild).toMatchSnapshot();
  });

  it('should render span tag', () => {
    const { container: { firstChild } } = render(
      <ChildrenWrapper submitting={true} />,
    );
    expect(firstChild.tagName).toEqual('SPAN');
  });

  it('should render content correctly', () => {
    const content = 'content';
    const { container: { firstChild } } = render(
      <ChildrenWrapper submitting={true}>
        {content}
      </ChildrenWrapper>,
    );
    expect(firstChild).toHaveTextContent(content);
    expect(firstChild).toMatchSnapshot();
  });
});
