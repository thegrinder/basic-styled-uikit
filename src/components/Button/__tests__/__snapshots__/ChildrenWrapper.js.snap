// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChildrenWrapper /> should render content correctly 1`] = `
<span
  class="ChildrenWrapper-sc-30uoyc-0 kEyivn"
>
  content
</span>
`;

exports[`<ChildrenWrapper /> should render correctly 1`] = `
<span
  class="ChildrenWrapper-sc-30uoyc-0 kEyivn"
/>
`;
